webapp-crud.go

```diff
@@ -1,38 +1,115 @@
 package main
 
 import (
-  "encoding/json"
-  "net/http"
-  "github.com/galuhest/item-crud-golang"
-  "github.com/julienschmidt/httprouter"
-  "log"
+    "fmt"
+    "strconv"
+    "net/http"
+    "github.com/galuhest/item-crud-go"
+    "github.com/julienschmidt/httprouter"
+    "log"
 )
 
 func main() {
-  router := httprouter.New()
-  router.GET("/", Index)
-  router.GET("/item/:id", Hello)
-  router.POST("/item", Hello)
-  router.POST("/item/:id", Hello)
-  router.POST("/item/:id/delete", Hello)
-
-  log.Fatal(http.ListenAndServe(":8080", router))
+    router := httprouter.New()
+    router.GET("/", Index)
+    router.GET("/item/:id", GetHandler)
+    router.POST("/item", CreateHandler)
+    router.POST("/item/:id", UpdateHandler)
+    router.POST("/item/:id/delete", DeleteHandler)
+
+    log.Fatal(http.ListenAndServe(":8080", router))
+}   
+
+// Halaman index untuk root
+func Index(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+    fmt.Fprint(w, "Welcome!\n")
+}
+
+// Handler untuk mengambil nama user
+func GetHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params)   {
+    temp := ps.ByName("id") 
+    id, err := strconv.Atoi(temp)
+    if err != nil {
+        log.Fatal("cant convert str to int")
+    }
+    db, err:= crud.ConnectDb()
+    if err != nil {
+        log.Fatal("cant open database")
+    }
+    defer db.CloseDb()
```

Comment:
nah instead of langsung di close, ini di close database connection pas aplikasi exit application saja, 

Dan result dari sisi webapp ini terima jadi sebagai result json (kalau protocol yang dipakai kemarin), cuman kok saya curiga kemarin pakai result json karena mau ditampilin di web. kalau saya setujunya sih mungkin dari item-crud-go dikembalikan dalam bentuk struct, dan nanti dari webapp nya bisa melakukan apapun yang diinginkan terhadap datanya, dan baru rendering/konversi ke data yang mau ditampilin oleh adalah responsibility dari webapp ini